### ISSUE TITLE
Huber-1992 data

### ISSUE THREAD

**Issue by LinguList** [glottobank/lexibank-data#21]
Currently being further curated by @thiagochacon (having some issues in accessing the online database), but potentially a nice thing to add.

We need still:

* glottolog-mappings

But the concepts are already there.

**Comment by thiagochacon** [comment]
I have the ISO mapping, but not glottolog. Would it help?

There you go

LINGUAISO-CODEAchaguaacaAwakwiBaniwabwiBarasanabsnBarimotBoraboaCabiyarícbbCamsákbhCarapanacbcCarijonacbdCentral TunebotufCha'palaaticbiChimilacbgCoguikogCubeocubCuibacuiCurripacokpcDesanodesDíminambpEmbera DariénempEmberá-CatíoctoEmberá-ChamícmiEmberá-TadótdcEpena Saija / Embera BasurudósjaGuahiboguhGuambianogumGuayaberoguoHupdajupIkaarhIngainbKakuacbvKoreguajecoeKotiriagvcMacaguán (Jitnu)mbnMacunamyyMaihikioreMirañamira1254MuinanebmrMurui HuitotohuuNukak MakúmbrOcainaocaPáezpbbPiapocopioPlayerogobPuinavepuiResígarorgrSaliba (Colombia)slcSecoyaseySionasnnSirianosriTanimuca-RetuarãtncTarianataeTatuyotavTotorottkTsafiqui pila cofTucanotuoTuyucatueWa'ikhanapirWaimahabaoWayuugucWitoto NɨpodehuxWoun MeunoaWuitotot MɨnɨcahtoYucunaycnYukpayupYurutíyui

Thiago Costa Chacon
Linguista – Professor Adjunto
Departamento de Linguística, Português e Línguas Clássicas
Universidade de Brasília


Date: Thu, 28 Jul 2016 03:42:00 -0700

**Comment by LinguList** [comment]
thanks, @thiagochacon , looks scrambled though: could you attach it later as spreadsheet?

**Comment by thiagochacon** [comment]
sorry, see if it is better now.



Thiago Costa Chacon
Linguista – Professor Adjunto
Departamento de Linguística, Português e Línguas Clássicas
Universidade de Brasília


Date: Thu, 28 Jul 2016 07:09:09 -0700

**Comment by LinguList** [comment]
did you attach it? Note that you can attach xlsx-files or other excel files by dragging them into the comment window (need to use the web-interface). 

@xrotwang: should we try and add Huber 1992? If so, I'd have the data already from the project with M. Cysouw, in a digitized form, also segmentized, but potentially a bit mis-parsed. I guess, despite the cool work that @thiagochacon has already been doing on searching and confirming cognates in the data, we'd do better in only listing lexemes + automatically inferred cognates, as the work by Thiago is still in progress and deserves a publication on its own. It'll take me a bit of time to look for the initial data, and prepare it, but I'd assign this threath then to me for the time being and try to make first arrangements during the next week, when I do not live in a tent anymore.

**Comment by xrotwang** [comment]
@LinguList yes, go ahead and do this. It would also be a nice example of dataset where cognate judgements get supplied in different dataset. I think we talked about this in a different issue, but I'd say this should best be handled via
- @thiagochacon getting his research published
- lexibank-data 
  - adding a dataset containing only cognate judgements
  - referencing the lexical data from Huber 1992

I don't know how much coordination we'd need to make sure the referencing can be done automatically. Both datasets would need to share lexeme IDs, I guess?

**Comment by LinguList** [comment]
yes, but this is something I can guarantee, as the webapp, where @thiagochacon is assigning cognates, is using a version that can be referenced, with links to the original dataset. In this case, the cognate judgments could also just relate to a subset of the data, depending on the work @thiagochacon wants to publish.

Problem is at the moment, that the web-app is facing difficulties, based on some mean JS issues, so I'll need to fix it at some point. But I'll just try and get the local identifiers in the data be in synch with any potential publication by @thiagochacon

**Comment by LinguList** [comment]
Just confirmed, that all QLC-identifiers which were produced in M. Cysouw's QLC project are in place in the data analyzed by @thiagochacon .

@thiagochacon , if you think that the data for even only a subset of the languages in huber is in a good state to be published (including cognate sets and alignments), we might consider doing that at some point. A publication could even just be made in a journal on datasets, so without analyses, only with data and some explanation on how it was produced (thinking of journals like [this](http://datascience.codata.org/) one). Or we could test the possibility of lexibank to offer exactly the possibility of this aspect of publishing some dataset, like cognate judgmnents and alignments with lexibank? Has this anywhere been planned so far?

**Comment by xrotwang** [comment]
@LinguList But the lexical data form Huber 1992 would still be a separate lexibank dataset, right? So we could start by putting this up and figure out how to model a "cognates-only dataset" in a second step?

**Comment by LinguList** [comment]
that's exactly, what I meant! I changed some parts of the representation, and @thiagochacon also corrected some wrong entries (due to bad OCR), but we kept all identifiers, and the QLC-Huber-1992 dataset is probably THE source we should use anyway, since it's the only available OCR. I'm still searching for the download link, as I have forgotten, where it can be found, but will post it, once I have it.

**Comment by LinguList** [comment]
Ah, there it is:

* http://www.quanthistling.info/data/downloads/csv/data.zip

So, what we probably need here, given the large amount of data, is a function to take data from QLC and download it and extract the relevant file. What I want to say is: given the large number of potentially useful dataset in QLC, we might not only restrict it to Huber1992. But I did not look too much at the other datasets, and many are not wordlists, but rather lexica. And we need to make sure we find a good way to quote the QLC project as the main provider, as Jelena Prokić and others have put much effort into its creation.

**Comment by LinguList** [comment]
Ah, and they list all iso-codes in the preamble of the datasets, so these will be convenient to use. Furthermore, Jelana made orthography profiles to segmentize the data, which will save us time to segment it later on.

**Comment by xrotwang** [comment]
For similar cases I have created the `pylexibank.providers` package. So I'd propose to add a `pylexibank.providers.qlc` module, providing
- filtered access to files in this zip-archive
- provenance information, i.e. a bibliographical citation, etc.

**Comment by LinguList** [comment]
sounds good to me. I just created a first huber-reed parser, of course without this providers-functionality. Should I still push it? Most of the important stuff is there, but I realized two things which make me worry:

* the file contains things called "ids" (the "qlcid") but these are not unique, since they use only concept_language as identifier, but there are synonyms. This is corrected with lingpy, but I need to get back to the web-app to make sure I created identical ids (usually this is the case, but I'm a bit too lazy to check it consistently now, yet it should be definitely done)
* I know they have page number and the like in the original data that was created, but I don't know why they are not in this subset of the data, which is a pity

A further problem, where I could search for a solution, but I guess you, @xrotwang can provide one quickly, is to go from iso-codes to glottolog. 

Should I still push this version, so that we have a first one upon we can then improve?

**Comment by xrotwang** [comment]
push! I'll have a look.

**Comment by LinguList** [comment]
done, I pushed directly...

**Comment by xrotwang** [comment]
@LinguList Please review and merge https://github.com/glottobank/lexibank-data/pull/31

**Comment by thiagochacon** [comment]
hi, I hope it is attached now.

I think it would be good to test the automatic cognate detection algorithm(s) and compare it with my manual work. As you will have entries co-referenced that would be a good study case for CALC.

As for publications, I need more time for curating the whole data set, but some language families or concepts are better than others. If we are going to publish as a data set at some point, it would be great to add language family names and language locations (I have that info too).

[huber lang names and iso code.txt](https://github.com/glottobank/lexibank-data/files/393755/huber.lang.names.and.iso.code.txt)

**Comment by LinguList** [comment]
sorry for late reply. I needed to do some training as long as the gym was empty. 

Reg. what @thiagochacon said, the info on language families and locations should come along automatically with glottolog, but we might want to review it to make sure all mappings are correct, given that you have created them independently.

Reg. publishing parts of the cognate sets: @thiagochacon, you might make a selection of a subset of languages where you think alignments and cognate sets are satisfying enough, and we could use this to write a short data-paper, including potentially a comparison with automatic cognate detection, a workflow, on how we used the curation tools, etc. It might also be a good example case for my EDICTOR tool, for which I should write an introductory paper anyway, so we could combine this. As to the third alternative, to directly publish it in lexibank, there is still the question whether we want to add that feature to lexibank (? or has it already been decided), and if yes, how we would proceed to make sure the things are nicely quotable.

**Comment by thiagochacon** [comment]
I am in for this, Mattis. 

Let's find some time in Jena to talk about that in detail.


Date: Mon, 1 Aug 2016 07:40:49 -0700
